sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/AmqpChannelProcessor.java

```diff
@@ -54,13 +52,15 @@
 
     public AmqpChannelProcessor(String fullyQualifiedNamespace, String entityPath,
```

Comment:
I actually wanted to change this API (it's in the `implementation` package) to not pass `ClientLogger`, but then it might cause some version conflicts between  SB/EH and core that otherwise would not happen. 

I think every class should create and own their logger instance (caller now passes name, which is also not grea). I've kept all key-value pairs intact though.

Reply:
> I think every class should create and own their logger instance (caller now passes name, which is also not grea). I've kept all key-value pairs intact though.

Yes, I generally prefer to have the ClientLogger created by the class that actually does the logging and not pass it around. The classname in the logs are indicative of which class did the logging. So, yes, if we can safely move away from not passing ClientLogger, then that would be ideal.

Reply:
I added another ctor without `ClientLogger` and deprecated the old one just in case of core-amqp and sdk version mismatch. We can remove the old overload after a few releases. I've also removed `ClientLogger` from `Handler` - it's only used in core-amqp, so there is no reason to keep the old constructor overload.